### PR TITLE
store/tikv: cache store.

### PR DESCRIPTION
**Pull request by disksing**
Cache `Store`s separately from regions, then we don't need to reload store's address from pd when leader changes or region splits.

cc @zimulala @shenli

### CODE REVIEW

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -269,21 +244,75 @@ func (c *RegionCache) loadRegion(bo *Backoffer, key []byte) (*Region, error) {
...
+}
+
+func (c *RegionCache) loadStoreAddr(bo *Backoffer, id uint64) (string, error) {
+	for {
+		store, err := c.pdClient.GetStore(id)
+		if err != nil {
+			err = errors.Errorf("loadStore from PD failed, id: %d, err: %v", id, err)
+			if err = bo.Backoff(boPDRPC, err); err != nil {
+				return "", errors.Trace(err)
+			}
```
Comment (zimulala) [inline store/tikv/region_cache.go]: Need add `continue` ?

File: store/tikv/region_cache.go [hunk 2]
```diff
@@ -55,28 +59,30 @@ type RPCContext struct {
...
-	defer c.mu.RUnlock()
-	if r, ok := c.mu.regions[id]; ok {
-		return &RPCContext{
-			Region: id,
-			KVCtx:  r.GetContext(),
-			Addr:   r.GetAddress(),
-		}
+	region, ok := c.mu.regions[id]
+	if !ok {
+		defer c.mu.RUnlock()
```
Comment (zimulala) [inline store/tikv/region_cache.go]: This `defer` useless.

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**